### DISCUSSION TITLE
用于 Deeplink 的单条标注外链方案

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3350] [服务]
> 在简悦中标注，然后将标注的 Deeplink 一键复制到任意双链笔记。

https://user-images.githubusercontent.com/81074/150922221-01aff575-7dc4-4807-a796-c11de6a57207.mp4

## 优势

跟简悦相比来说，大部分的 App 使用了 `URL Scheme` 方案，因此只能在 Mac 上使用，且需要安装对应 App，不利于嵌入其中双链笔记的「传播」，而简悦的方式拥有如下优势：

1. 不依赖于 `URL Scheme` 方案，而使用了 `http` or `https` 标准方案。
2. 无论是**内部链接** or **外部链接**，对应的都是你本地的文件，可以做到真快照。

## 使用前提

需要开启自定义 Markdown 模板，位置在选项页 → 服务 → 定制导出

![image](https://user-images.githubusercontent.com/81074/189465822-443fad23-f7d4-4980-b443-c0da66e103ee.png)

## 配置方案一
 
利用标注工具条的导出功能， 与 [浮动工具条的导出功能](https://github.com/Kenshin/simpread/discussions/1851) 操作类似，使用 [此模板](https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-2032887)，并复制到下图所示位置。

![image](https://user-images.githubusercontent.com/81074/150768975-f960456d-c060-470d-ae23-43706735c988.png)

### 配置浮动工具条的导出

> 位置在选项页 → 高级设定 → 标注

![image](https://user-images.githubusercontent.com/81074/119213528-3b057f00-baf2-11eb-80e3-c23899220924.png)

### 如何使用

在任意标注上使用下面的方式

![image](https://user-images.githubusercontent.com/81074/150770090-7a77f266-98c8-4a1f-8995-0763c7ae202f.png)

## 配置方案二

利用标注工具条的外链功能，选项页 → 高级设定 → 标注，使用 [此模板](https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-2039573)，并复制到下图所示位置

![image](https://user-images.githubusercontent.com/81074/150897881-d2e64df2-2147-4e91-945f-0e94cdadd601.png)

### 如何使用

在任意标注上使用下面的方式

![image](https://user-images.githubusercontent.com/81074/150898291-f54370e2-5d48-4d2a-ac90-409c2cdacdcd.png)

## 方案一和二的区别

方案一「占用」了导出单条标注时的模板，所以建议使用**方案二**的方式，但方案一有更多定制模板样式的空间。

## 引申

1. 需要配置 [同步助手](https://github.com/Kenshin/simpread/discussions/2754)。
2. 使用内部链接的话，仅需配置 [知识库](https://github.com/Kenshin/simpread/discussions/2221) 即可。
3. 只有使用外部永久链接才需要配置 [资料库](https://github.com/Kenshin/simpread/discussions/2956) 功能。

## 关联

- [利用好浮动标注栏的导出功能](https://github.com/Kenshin/simpread/discussions/1851)

**Comment by XMB-7** [comment]
请问该功能是否只能跳转到文章的开头, 亦即不支持自动跳转到标注的所在位置? 
参见上面演示视频第5s往后, 还要滑动鼠标滚轮, 自己往下寻找该条标注. 
如果是的话, 能否考虑联动浏览器插件link to text fragment, 使插入的模板链接能实现链接跳转之后的自动定位.

**Reply by Kenshin** [reply]
标注的内部链接是个锚点，点击后肯定要跳转到标注对应的位置，不会只在页面的开头，所以正常不会出现你说的问题。（或者我没有理解你的问题，方便的话，请新建一个 Issues 描述，包括给出视频参考。

> link to text fragment

其实并不好用，因为出现「标记」后，点击页面后就消失了。另外，扩展端也有这个功能，甚至于 [Markdown 辅助增强](https://github.com/Kenshin/simpread/discussions/3725) 内置了跟此一样的方式，占位符叫 `{{an_org_uri}}`。

**Reply by XMB-7** [reply]
呃...你的意思是, 在本页页首的展示视频里, 第5秒打开链接, 然后第6秒跳转到浏览器并翻滚到对应标注所在位置, 是自动完成的, 而非手动下滑鼠标滚轮才完成定位的?
但是我这边试了一下, 好像有点问题. 跳转只能跳转到页首, 未能定位到页尾的那个标注.
![image](https://user-images.githubusercontent.com/65625632/167239922-2d3fae61-4996-4dd2-8219-85ffd45476fd.png)
![image](https://user-images.githubusercontent.com/65625632/167239965-45b77fc2-26b3-4389-9bc7-6ab5b3f84173.png)
![image](https://user-images.githubusercontent.com/65625632/167243222-4198d02a-ae12-46d0-ab84-1fdfc47b11e8.png)

不知道是不是因为用了外部链接, 我这边再照着各个文档多尝试一下吧. 话说双链的文档读着着实有点费劲, 经常一个点跳另一个点跳另一个点, 读着读着就晕了...

**Reply by Kenshin** [reply]
> 你的意思是, 在本页页首的展示视频里, 第5秒打开链接, 然后第6秒跳转到浏览器并翻滚到对应标注所在位置, 是自动完成的

是的。

***

建议先以内部链接的方案测试，外部链接下周有个新的方式。

外部链接也就是阅读列表的功能，配置教程[在这里](https://kb.simpread.pro/#/page/%E9%98%85%E8%AF%BB%E5%88%97%E8%A1%A8)，此功能是个复合型功能，大致需要下面截图的内容才能完成，不过我看你（的截图）差不多折腾的没问题了。

![image](https://user-images.githubusercontent.com/81074/167243288-72520d63-bb4a-43f8-aa8d-120389833f7f.png)

**Reply by XMB-7** [reply]
发现问题了, 要调整两个方案的设置.

![image](https://user-images.githubusercontent.com/65625632/167257186-d69a0aee-872f-4239-8c67-ef1de1d77a68.png)
---
![image](https://user-images.githubusercontent.com/65625632/167257202-b499fea0-b141-4467-8aea-41ea8126ed1c.png)

当下图的红框设置得跟上图的红框一致的时候, 就会导致我所说的这个问题.
要么直接把单条标注的定制化模板关了, 要么把单条标注的定制化模板改了, 比如删掉(<{{an_ext_uri}}>), 然后一切功能就都正常了.

**Reply by Kenshin** [reply]
okay 搞定就好，我也研究下这个问题。
